**<sub><sub>![P1 Badge](https://img.shields.io/badge/P1-orange?style=flat)</sub></sub>  Declare design tokens module for default theme**

**<sub><sub>![P1 Badge](https://img.shields.io/badge/P1-orange?style=flat)</sub></sub>  Declare design tokens module for default theme**

The new default theme now imports `@dynui/design-tokens`, but this package is not declared as a dependency of the root workspace and no TS path alias is provided. Running `pnpm exec tsc -p tsconfig.json` fails with `TS2307: Cannot find module '@dynui/design-tokens'` (and the subsequent `value is of type 'unknown'` errors are a side effect). Without adding the workspace dependency or a compile-time alias, the project cannot type-check or build.

Useful? React with 👍 / 👎.

_Originally posted by @chatgpt-codex-connector[bot] in https://github.com/mgasic/dynui-fixed/pull/157#discussion_r2463951906_

--- a/packages/design-tokens/src/css-generator.ts
+++ b/packages/design-tokens/src/css-generator.ts
@@ -11,9 +11,9 @@ function sanitizeTokenKey(key: string): string {
   return key
     .trim()
     .toLowerCase()
-    .replace(/[^a-z0-9_-]/g, '-')
-    .replace(/-+/g, '-')
-    .replace(/^-|-$/g, '')
+    .replace(/[^a-z0-9_-]/g, '-') // Replace periods and other invalid chars with dashes
+    .replace(/-+/g, '-')  // Collapse multiple dashes
+    .replace(/^-|-$/g, '') // Remove leading/trailing dashes
 }
 
 function normalizeValue(value: TokenPrimitive | readonly TokenPrimitive[]): string {

